src/types.rs

```diff
@@ -909,14 +1129,114 @@ impl Term {
         }
     }
     /// Given a vector of contraints, return a substitution which satisfies the constrants.
-    /// If the constraints are not satisfiable, `None` is retuned. Constraints are in the form of
+    /// If the constraints are not satisfiable, return `None`. Constraints are in the form of
     /// patterns, where substitutions are only considered for variables in the first term of each
     /// pair.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use term_rewriting::{Signature, Term, parse_term};
+    /// # use std::collections::{HashMap, HashSet};
+    ///
+    /// let mut sig = Signature::default();
+    /// let x = sig.new_var(Some("x_".to_string()));
+    /// let y = sig.new_var(Some("y_".to_string()));
+    /// let s = sig.new_op(1, Some("S".to_string()));
+    /// let a = sig.new_op(1, Some("A".to_string()));
+    /// let b = sig.new_op(0, Some("B".to_string()));
+    ///
+    /// let args = vec![];
+    /// // t is equivalent to S(A)
+    /// let t = Term::Application {
+    ///     op: s,
+    ///     args: vec![Term::Application { op: a, args }],
+    /// };
+    /// // t2 is equivalent to S(x)
+    /// let t2 = Term::Application {
+    ///     op: s,
+    ///     args: vec![Term::Variable(x)],
+    /// };
+    /// // t3 is equivalent to S(y)
+    /// let t3 = Term::Application {
+    ///     op: s,
+    ///     args:vec![Term::Variable(y)],
+    /// };
+    /// // t4 is equivalent to A(x)
+    /// let t4 = Term::Application {
+    ///     op: a,
+    ///     args:vec![Term::Variable(x)],
+    /// };
```

Comment:
Rather than constructing the terms manually, could you use `parse_term` to make this example and the `unify` example shorter?